clients/src/main/java/org/apache/kafka/clients/consumer/internals/Coordinator.java

```diff
@@ -460,6 +460,12 @@ private void maybeAutoCommitOffsetsSync() {
                     offsetAndMetadata.offset(), offsetAndMetadata.metadata()));
         }
 
+        // reset generation/consumerId to default values if using manual assignment
+        if (!subscriptions.partitionsAutoAssigned()) {
+            this.consumerId = OffsetCommitRequest.DEFAULT_CONSUMER_ID;
```

Comment:
As for whether it's necessary to have both UNKNOWN_CONSUMER_ID and DEFAULT_CONSUMER_ID, I don't have a strong preference. There doesn't seem to be a convention in place to store default values which are shared across multiple request types. Perhaps they should go in `Protocol`? I'm probably inclined to leave it as it is, but at least we can make Coordinator only depend on one of the values.